Add Select component

venmovs opened the pull request:
Пришлось навертеть кастомный select, так как не получилось стилизовать дефолтный.
При необходимости добавлю переключение в выпадающем списке через tab.

Впервые трогаю tailwind, буду благодарен подсказкам и замечаниям 😊

Closes #44

venmovs committed:
create Select component

venmovs committed:
add new props for Select component

venmovs committed:
add styles

venmovs committed:
merge updates

YurikEz committed:
add ui Components - Button, Title, Subtitle, Text, Description, Link, Input (#48)

* review

* review

* review

* add ui component Button

* review

* add ui component - Button.tsx

Closes #45

* updated ui component - Button.tsx

* add ui components - Title.tsx, Subtitle.tsx

Closes #43, Closes #42

* updated ui components - Title.tsx, Subtitle.tsx

Closes #43, Closes #42

* add empty line in end

* review ui component Title.tsx

* add ui components Text.tsx, Description.tsx, Link.tsx

Closes #49, Closes #50, Closes #51

* updated ui component Link.scss

* Up classes, tailwind

* Up input styles

* review

* lint

* fix review

* Ревью по комментам от ментора

- Вынес дефолтные классы вне функций компонентов в константы
- Заюзал useMemo
- Создал мелкие Element'ы для некоторых компонентов
- Ревью
- Переписал Link
- Вынес меню только на главную страницу

* lint

* lint

* Up undefined

* Up undefined

* back menu to dashboard

Co-authored-by: Елманов Юрий Алексеевич <yuelmanov@phoenixit.ru>
Co-authored-by: tenutes <leopard19670508>

venmovs committed:
add Dashboard

venmovs committed:
Merge branch 'sprint-5-45-44' of https://github.com/YaFriends/ya-game-frontend into sprint-5-45-44

venmovs committed:
change select tag to custom select

venmovs committed:
update styless and start create placeholder

venmovs committed:
Select is ready

YurikEz commented on src/components/ui/Select/Select.tsx:
```diff
@@ -0,0 +1,96 @@
+import React, { useState, useEffect, useRef, FC } from 'react';
+
+import { Arrow } from '../../../images/icons/Arrow';
+import { Label } from '../Input/ElementLabel';
```
Этот Label принадлежит только к Input
Скорее всего придется дублировать подобный элемент тебе в селект в папочку

venmovs replied:
Может быть стоит его вынести в отдельный компонент и директорию?

YurikEz replied:
Ну в целом, давай создадим еще один компонент для Label на подобии
А Input я его потом подключу, когда выкатим наши ПР в прод, чтобы не решать этот конфликт сразу

venmovs replied:
Добавил компонент Label, в Input пока ничего не менял

YurikEz commented on src/components/ui/Select/Select.tsx:
```diff
@@ -0,0 +1,96 @@
+import { Label } from '../Input/ElementLabel';
+
+import { Option } from './Option';
+import { SelectProps } from './SelectTypes';
```
Думаю все типы можно описать тут
сделать export и если они нужны вызвать их в option 
тк мы ранее договорились, что будем хранить типы прям в файле

venmovs replied:
Вынес все типиы сюда, файл с типами удалил

YurikEz commented on src/components/ui/Select/Option.tsx:
```diff
@@ -0,0 +1,30 @@
+import React, { FC, PropsWithChildren } from 'react';
+
+import { OptionsProps, SelectedChangeType } from './SelectTypes';
```
Вызвать типы из Select.tsx

venmovs replied:
сделал

YurikEz commented on src/components/ui/Select/Select.tsx:
```diff
@@ -0,0 +1,96 @@
+    if (option.value === selected.value) {
+      return null;
+    }
+    const { value, label } = option;
```
эту строку `const { value, label } = option;` убрать

venmovs replied:
сделал

YurikEz commented on src/components/ui/Select/Select.tsx:
```diff
@@ -0,0 +1,96 @@
+    setOpen(prevState => !prevState);
+  };
+
+  const renderedOptions = options.map(option => {
```
а тут `option` превратим `({ value, label })`
те сразу сделаем диструкторизацию

venmovs replied:
сделал

YurikEz commented on src/components/ui/Select/Select.tsx:
```diff
@@ -0,0 +1,96 @@
+  );
+};
+
+export { Select };
```
`export { Select };` думаю это можно убрать 
и выше просто сразу возвращать Select

YurikEz commented on src/components/ui/Select/Select.tsx:
```diff
@@ -0,0 +1,96 @@
+import { Option } from './Option';
+import { SelectProps } from './SelectTypes';
+
+const Select: FC<SelectProps> = ({ label, options, selected, onSelectedChange, placeholder }) => {
```
Вот тут до `const Select: FC`  добавить `export`

YurikEz commented on src/components/ui/Select/SelectTypes.ts:
```diff
@@ -0,0 +1,18 @@
+import { SelectHTMLAttributes, SetStateAction } from 'react';
+
+type SelectedChangeType = (option: SetStateAction<OptionsProps>) => void;
```
Вынести весь этот файл в файл `Select.tsx`

YurikEz commented on src/images/icons/Arrow.tsx:
```diff
@@ -0,0 +1,36 @@
+  degree?: string;
+}
+
+const Arrow: FC<ArrowProps> = ({ degree = '0' }) => {
```
Добавить `export`

venmovs replied:
сделал

YurikEz commented on src/images/icons/Arrow.tsx:
```diff
@@ -0,0 +1,36 @@
+  );
+};
+
+export { Arrow };
```
УБрать `export { Arrow };`

venmovs replied:
сделал

YurikEz commented on src/components/ui/Select/Option.tsx:
```diff
@@ -0,0 +1,30 @@
+  );
+};
+
+export { Option };
```
УБрать `export { Option };`

venmovs replied:
сделал

YurikEz commented on src/components/ui/Select/Option.tsx:
```diff
@@ -0,0 +1,30 @@
+
+import { OptionsProps, SelectedChangeType } from './SelectTypes';
+
+const Option: FC<
```
Добавить `export` перед

venmovs replied:
сделал

YurikEz commented on src/components/ui/Select/Option.tsx:
```diff
@@ -0,0 +1,30 @@
+    'last:h-[50px]',
+  ];
+  return (
+    <div className={DEFAULT_CLASSES.join(' ')} onClick={() => onSelectedChange(option)}>
```
А тут это `onClick={() => onSelectedChange(option)}`, если мы изменим на такое 
`onClick={onSelectedChange(option)}` по идее так же должно работать

venmovs replied:
вроде нельзя передавать параметр без callback, только если там не event ожидается, мб ошибаюсь

YurikEz replied:
Да, ты прав

YurikEz reviewed (commented):
Хорошая работа, чуть чуть подправить и будет супер

venmovs committed:
changed export and file structure

YurikEz committed:
review components & add Login layout (#52)

* review

* review

* review

* add ui component Button

* review

* add ui component - Button.tsx

Closes #45

* updated ui component - Button.tsx

* add ui components - Title.tsx, Subtitle.tsx

Closes #43, Closes #42

* updated ui components - Title.tsx, Subtitle.tsx

Closes #43, Closes #42

* add empty line in end

* review ui component Title.tsx

* add ui components Text.tsx, Description.tsx, Link.tsx

Closes #49, Closes #50, Closes #51

* updated ui component Link.scss

* Up classes, tailwind

* Up input styles

* review

* review

* add 100% width for Input.tsx

* review link

* add historyApiFallback

* updated Button.tsx

* Add Login.tsx layout

* review className

* lint

* fix review

* Ревью по комментам от ментора

- Вынес дефолтные классы вне функций компонентов в константы
- Заюзал useMemo
- Создал мелкие Element'ы для некоторых компонентов
- Ревью
- Переписал Link
- Вынес меню только на главную страницу

* lint

* lint

* Up undefined

* Up undefined

* back menu to dashboard

* review style

* fix names

* rename handle click

* rename class & update translation.ts

* add ui-component Form.tsx

* review code

Co-authored-by: Елманов Юрий Алексеевич <yuelmanov@phoenixit.ru>
Co-authored-by: tenutes <leopard19670508>

YurikEz committed:
add Register.tsx layout (#53)

* review

* review

* review

* add ui component Button

* review

* add ui component - Button.tsx

Closes #45

* updated ui component - Button.tsx

* add ui components - Title.tsx, Subtitle.tsx

Closes #43, Closes #42

* updated ui components - Title.tsx, Subtitle.tsx

Closes #43, Closes #42

* add empty line in end

* review ui component Title.tsx

* add ui components Text.tsx, Description.tsx, Link.tsx

Closes #49, Closes #50, Closes #51

* updated ui component Link.scss

* Up classes, tailwind

* Up input styles

* review

* review

* add 100% width for Input.tsx

* review link

* add historyApiFallback

* updated Button.tsx

* Add Login.tsx layout

* review className

* lint

* fix review

* Ревью по комментам от ментора

- Вынес дефолтные классы вне функций компонентов в константы
- Заюзал useMemo
- Создал мелкие Element'ы для некоторых компонентов
- Ревью
- Переписал Link
- Вынес меню только на главную страницу

* lint

* lint

* Up undefined

* Up undefined

* back menu to dashboard

* review style

* add Register.tsx page layout

* fix names

* fix

* rename handle click

* rename class & update translation.ts

* add ui-component Form.tsx

* fix review

* review code

* review

Co-authored-by: Елманов Юрий Алексеевич <yuelmanov@phoenixit.ru>
Co-authored-by: tenutes <leopard19670508>

venmovs committed:
merging master and resolved conflicts

venmovs committed:
create Label component

venmovs committed:
linter fixes

Tenutes commented on src/components/ui/Select/Select.tsx:
```diff
@@ -0,0 +1,120 @@
+
+export interface SelectProps extends SelectHTMLAttributes<HTMLSelectElement> {
+  label: string;
+  options: OptionsProps[];
```
ЯБ назвал OptionProps, т.к.  это единственная Option, а не мульти какая-то

venmovs replied:
исправил)

Tenutes commented on src/components/ui/Select/Select.tsx:
```diff
@@ -0,0 +1,120 @@
+      return;
+    }
+    const target = event.target as HTMLElement;
+    if (dropDownRef.current && dropDownRef.current.contains(target)) {
```
dropDownRef.current?.contains

venmovs replied:
предохранился))

Tenutes replied:
Там можешь первое условие `dropDownRef.current` вообще убрать, и оставить лишь второе с contains и все, мона лить. Красавчик)

venmovs committed:
fix naming and add optional chaning operator

YurikEz reviewed (approved):
Красиво!

venmovs committed:
removed re-check
